### PR TITLE
Fix up docker build scripts

### PR DESCRIPTION
**Pull request by adleong**
A few small improvements to our docker build scripts:

* Centralized the list of docker images to a DOCKER_IMAGES variable defined in _docker.sh
* Build scripts now honor the TAG variable, if defined
* Unused docker-images script has been removed

We also update the `--control-plane-version` Linkerd install flag to affect the policy controller version as well.

Taken together, this enables the following workflow for building and deploying changes to individual Linkerd components.  For example, suppose you wish to deploy changes which only affect the controller image:

```console
# Begin by building all images at main with a dev tag
> TAG=alex-dev bin/docker-build

# OR begin by retagging all images from a recent release
> bin/docker-retag-all edge-21.8.4 alex-dev

# Make changes and then rebuild specific component
> TAG=alex-dev bin/docker-build-controller

# Load images into kind
> TAG=alex-dev bin/image-load --kind --cluster alex

# Install Linkerd
> bin/linkerd install --control-plane-version alex-dev --proxy-version alex-dev | k apply -f -
```

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

File: bin/image-load [hunk 1]
```diff
@@ -37,7 +40,7 @@ usage() {
 
 while :
 do
-  case $1 in
+  case ${1:-} in
```
Comment (kleimkuhler) [inline bin/image-load:43]: What was the reasoning for this change?

**Review by kleimkuhler — APPROVED** [review]
Looks awesome! Thanks for the instructions in the description they work well. This is going to be handy for making changes to certain components and not having to rebuild everything 🤙.

File: bin/image-load (hunk 1 above)
Comment (adleong) [reply bin/image-load:43]: We're sourcing _docker.sh which contains `set -u` which causes us to throw errors for unbound variables.  When we get to the end of the argument list `$1` is unbound so with `-u` set, we throw an error here.  Instead, I change it to default to empty so avoid the unbound variable error.

Comment (kleimkuhler) [reply bin/image-load:43]: Ah okay thanks for the explanation!

**Review by alpeb — APPROVED** [review]
:cool: !

**Pull request merged**